### PR TITLE
Quick sort

### PR DESCRIPTION
**Pull request by steventhan**

### CODE REVIEW

File: src/quick_sort.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        else:
+            left.append(a_list[1])
+            right.append(a_list[0])
+    else:
+        hi = a_list[-1]
+        low = a_list[0]
+        midpoint = a_list[len(a_list)//2]
+        pivot = sorted([low, hi, midpoint])[1]
+        for item in a_list:
+            if item < pivot:
```
Comment (WillWeatherford) [inline src/quick_sort.py]: This will work in a lot of cases, but there is another scenario which you are not considering, and for which you need to test. Currently your algorithm is getting stuck in an infinite loop.

Make sure you include some test cases with duplicate items in your list to be sorted.
